runtime/datastructures/src/decls.rs

```diff
@@ -0,0 +1,70 @@
+extern crate im;
+
+use decls::im::hashmap::HashMap;
+use decls::im::hashset::HashSet;
+use decls::im::vector::Vector;
+
+pub enum Block {}
+#[allow(non_camel_case_types)]
+pub enum mp_limb_t {}
+
+#[repr(C)]
+#[allow(non_camel_case_types)]
+pub struct Int(
+  i32, // _mp_alloc
+  i32, // _mp_size
+  *const mp_limb_t, // _mp_d
+);
+
+pub type K = *const Block;
+pub type Map = HashMap<K, K>;
+pub type Set = HashSet<K>;
+pub type List = Vector<K>;
```

Comment:
Let's see how it goes, but based on the documentation the minimum size of this Vector is a bit high:
https://docs.rs/im/11.0.1/im/vector/index.html
Perhaps 2 or 3 of those arrays will often be shared for small vectors, and we could add further small-vector optimizations if needed.